### PR TITLE
added three utility macros

### PR DESCRIPTION
**Pull request by andrewhong5297**
Brief comments on the purpose of your changes:

Spoke with @soispoke about this earlier, created two macros I want as UDFs for getting the symbol/prices_symbol/prices_address or block explorer base url for a given chain. I also added an href creator so people can stop doing the long `concat <a href = "" target='_blank'> </a>` thing.

Not sure if these need tests or how to test them.

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributors)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by henrystats** [comment]
this is amazingggg and going to be really helpful, thank you serrr!

**Comment by jeff-dude** [comment]
did we determine how these can be used post-merge? i was thinking the macro will exist in spellbook, but can only referenced in future spell models, not necessarily directly in the db (i.e. when querying dune)

**Comment by andrewhong5297** [comment]
> did we determine how these can be used post-merge? i was thinking the macro will exist in spellbook, but can only referenced in future spell models, not necessarily directly in the db (i.e. when querying dune)

@soispoke mentioned the macro would be usable in the query interface too

**Comment by andrewhong5297** [comment]
Ah also not by computer but the explorer url macro should only return url, none of the other columns

**Comment by sohwak** [comment]
I happened to see this post, and I also think it would be very useful. I was just dealing with the price of the native tokens on the multi-chain spells, so I am more interested.

I looked at the tokens that were added, and all tokens used as alternatives in other native tokens are the same as mine except for the ones in Polygon. I was referring to MATIC `0x0000000000000000000000000000001010` for the native token on Polygon. I was curious and looked up WMATIC `0x0d500b1d8e8ef31e21c99d1db9a6444d3adf1270` this PR mentioned, but it is not currently in `prices.usd`.

So I think it would be better to use MATIC for Polygon.

**Comment by andrewhong5297** [comment]
Makes sense to me!

**Comment by andrewhong5297** [comment]
okay fixes are in, this should be ready to merge or test now

File: macros/public/get_chain_native_token.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{% macro get_chain_native_token(chain, column) %}
+   SELECT
```
Comment (jeff-dude) [inline macros/public/get_chain_native_token.sql:2]: will this fail with two selects?

**Comment by jeff-dude** [comment]
if you want to test pre-merge, you can add sql files under ```tests/macros/public/``` which call each macro within a query. can be a basic query, but you'd at least see results and can force result to return nothing to pass -- i.e. filter out expected result

File: macros/public/get_chain_native_token.sql (hunk 1 above)
Comment (andrewhong5297) [reply macros/public/get_chain_native_token.sql:2]: haha yes oops

**Comment by andrewhong5297** [comment]
can't seem to call them inside tests.

https://github.com/duneanalytics/spellbook/actions/runs/3421135094/jobs/5696864292

Screen Shot 2022-11-08 at 12 12 44 PM

**Comment by andrewhong5297** [comment]
Okay, the three tests are passing now!

**Review by jeff-dude — APPROVED** [review]
nice work!

**Comment by dot2dotseurat** [comment]
@andrewhong5297 looks like this is causing issues for `dbt compile`. I'm going to revert and see what we can do to exclude this automatically from compilation.

**Pull request merged**